File: app/actions/bulk-cp.ts
import * as csv from 'csv-writer';
import Stripe from 'stripe';

const stripe = new Stripe('pass_your_stripe_secret_key');

async function generateCoupons(numCoupons: number): Promise<void> {
    const coupons = [];
    
    for (let i = 0; i < numCoupons; i++) {
        const couponId = `UniqueCoupon_${i+1}`;
        try {
            const coupon = await stripe.coupons.create({
                name: couponId,
                duration: 'once',
                percent_off: 20,
                currency: 'usd'
            });
            
            coupons.push({ 'Stripe ID': coupon.id, 'Custom ID': couponId });
            console.log(`Generated coupon ${couponId} with ID ${coupon.id}`);
        } catch (error) {
            console.error(`Error generating coupon ${couponId}:`, error);
        }
    }

    const csvWriter = csv.createObjectCsvWriter({
        path: 'bulk_coupons.csv',
        header: [
            { id: 'Stripe ID', title: 'Stripe ID' },
            { id: 'Custom ID', title: 'Custom ID' }
        ]
    });
    
    await csvWriter.writeRecords(coupons);
    
    console.log(`A total of ${coupons.length} coupons have been generated and saved to bulk_coupons.csv.`);
}

generateCoupons(5);

File: bulk-cp.py
import csv
import stripe

stripe.api_key = 'pass_your_stripe_secret_key'

def generate_coupons(num_coupons):
    coupons = []
    
    for i in range(num_coupons):
        coupon_id = f"UniqueCoupon_{i+1}"  
        coupon = stripe.Coupon.create(
            name=coupon_id,
            duration='once',
            percent_off=20,  
            currency='usd',
        )
        coupons.append([coupon.id, coupon_id])
        print(f"Generated coupon {coupon_id} with ID {coupon.id}")

    with open('bulk_coupons.csv', 'w', newline='') as csvfile:
        writer = csv.writer(csvfile)
        writer.writerows(coupons)

    print(f"A total of {len(coupons)} coupons have been generated and saved to bulk_coupons.csv.")

generate_coupons(5)